### PR TITLE
[Test] Validate the /config endpoint of RESTful module

### PR DESCRIPTION
**Pull request by sayaleeraut**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by MohitBis — APPROVED** [review]
LGTM

**Comment by sayaleeraut** [comment]
Execution logs - https://ocs3xstage-jenkins-csb-rhgsocs3x.apps.ocp-c1.prod.psi.redhat.com/view/Executor/job/qe-squid-test-executor-openstack/26/console

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    # Declare the URL for the request
+    url = f"https://{mgr_ip}:8003"
+
+    if method == "GET":
+        return _get(config, url, endpoint, username, key, status_code)
+
+    elif method == "PATCH":
+        return _patch(config, url, endpoint, username, key, status_code)
+    return 0
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_config_endpoint.py:83]: this code is not reachable

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+def run(ceph_cluster, **kw):
+    config = kw.get("config", {})
+
+    # Get the MGR node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+    mgr_ip = node.ip_address
+
+    # Read the test configs
+    username = config.get("username")
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_config_endpoint.py:54]: better to have single check -
```
if not (username and endpoint and method and status_code):
    raise ....
```

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+def _get(config, url, endpoint, username, key, status_code):
+    # Check if any <args> are present
+    if config.get("args"):
+        endpoint += f"/{config.get('args')}"
+    _code, _ = Config(url=url, api=endpoint).get(
+        username=username, key=key, check_sc=True
+    )
+    if _code == status_code:
+        return 0
+    raise OperationFailedError(
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_config_endpoint.py:26]: Move this exception to `run` method

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/mgr/test_restful_operations_config_endpoint.py (hunk 2 above)
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_config_endpoint.py:46]: Done, please check L50

File: tests/mgr/test_restful_operations_config_endpoint.py (hunk 1 above)
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_config_endpoint.py:75]: Done, made correction

File: tests/mgr/test_restful_operations_config_endpoint.py (hunk 3 above)
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_config_endpoint.py]: Done

**Review by pranavprakash20 — APPROVED** [review]
LGTM

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            raise OperationFailedError(
+                f"RESTful GET method for {endpoint} failed with status code:{out}"
+            )
+
+    elif method == "PATCH":
+        out = _patch(config, url, endpoint, username, key)
+        if out != status_code:
+            raise OperationFailedError(
+                f"RESTful PATCH method for {endpoint} failed with status code:{out}"
+            )
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_config_endpoint.py:74]: Add extra line

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            raise ConfigError(CONFIG_ERR_MSG.format(_config))
+
+    # Get key for API user
+    key = json.loads(CephAdm(node).ceph.restful.list_key()).get(username, None)
+    if not key:
+        raise OperationFailedError(f"Failed to get key for API user {username}")
+
+    # Declare the URL for the request
+    url = f"https://{mgr_ip}:8003"
+
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_config_endpoint.py:61]: Add comment

File: tests/mgr/test_restful_operations_config_endpoint.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    # Get the MGR node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+    mgr_ip = node.ip_address
+
+    # Read the test configs
+    username = config.get("username")
+    endpoint = config.get("endpoint")
+    method = config.get("method")
+    status_code = config.get("status_code")
+    for _config in [username, endpoint, method, status_code]:
```
[the remark this answers is not in the document]
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_config_endpoint.py:50]: I added it to get the exact missing config information through error msg in L52

**Comment by sayaleeraut** [comment]
Execution logs -
https://ocs3xstage-jenkins-csb-rhgsocs3x.apps.ocp-c1.prod.psi.redhat.com/view/Executor/job/qe-squid-test-executor-openstack/36/console 

ALL tests are passing.

**Review by vamahaja — APPROVED** [review]

**Pull request merged**